src/xkbcomp/symbols.c

```diff
@@ -61,6 +61,30 @@
 #include "include.h"
 #include "keysym.h"
 
+
+// TODO: how to convert warnings to program errors?
```

Comment:
you can't, not in the library. It's up to the caller to detect a log call with `XKB_LOG_LEVEL_ERROR` and convert that to a program error.

Though the problem with that is that you can't differ between a generic error that occurred (e.g. "cant' find file") and a parser error that you specifically enabled.